src/hamerly.jl

```diff
@@ -115,52 +117,55 @@ function create_containers(alg::Hamerly, X, k, nrow, ncol, n_threads)
     )
 end
 
+
 """
-    chunk_initialize(alg::Hamerly, containers, centroids, design_matrix, r, idx)
+    chunk_initialize(alg::Hamerly, containers, centroids, X, weights, metric, r, idx)
 
 Initial calulation of all bounds and points labeling.
 """
-function chunk_initialize(alg::Hamerly, containers, centroids, X, weights, r, idx)
+function chunk_initialize(alg::Hamerly, containers, centroids, X, weights, metric, r, idx)
     T = eltype(X)
     centroids_cnt = containers.centroids_cnt[idx]
     centroids_new = containers.centroids_new[idx]
 
     @inbounds for i in r
-        label = point_all_centers!(containers, centroids, X, i)
+        label = point_all_centers!(containers, centroids, X, i, metric)
         centroids_cnt[label] += isnothing(weights) ? one(T) : weights[i]
         for j in axes(X, 1)
             centroids_new[j, label] += isnothing(weights) ? X[j, i] : weights[i] * X[j, i]
         end
     end
 end
 
+
 """
-    update_containers(::Hamerly, containers, centroids, n_threads)
+    update_containers(::Hamerly, containers, centroids, n_threads, metric)
 
 Calculates minimum distances from centers to each other.
 """
-function update_containers(::Hamerly, containers, centroids, n_threads)
+function update_containers(::Hamerly, containers, centroids, n_threads, metric)
     T = eltype(centroids)
     s = containers.s
     s .= T(Inf)
     @inbounds for i in axes(centroids, 2)
         for j in i+1:size(centroids, 2)
-            d = distance(centroids, centroids, i, j)
-            d = T(0.25)*d
+            d = distance(metric, centroids, centroids, i, j)
```

Comment:
Can we write them in a single line? What's the point of splitting in two different lines?